python/lsst/ts/ess/csc/air_turbulence_accumulator.py

```diff
@@ -0,0 +1,177 @@
+# This file is part of ts_ess_csc.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AirTurbulenceAccumulator"]
+
+import math
+from collections.abc import Sequence
+
+import numpy as np
+
+
+class AirTurbulenceAccumulator:
+    """Accumulate air turbulence data from a 3-d anemometer.
+
+    This supports writing the airTurbulence telemetry topic,
+    whose fields are statistics measured on a set of data.
+
+    Parameters
+    ----------
+    num_samples : `int`
+        The number of samples to read before producing aggregate data.
+
+    Attributes
+    ----------
+    num_samples : `int`
+        The number of samples to read before producing aggregate data.
+    timestamp : list[float]
+        List of timestamps (TAI unix seconds)
+    speed : list[float]
+        List of wind speed in x, y, z (km/s)
+    sonic_temperature : list[float]
+        List of sonic temperature (deg C)
+    num_bad_samples : int
+        Number of invalid samples.
+
+    Notes
+    -----
+    *To Use*
+
+    For each data sample read, call ``add_sample``.
+    Then check ``do_report``, and if true, call ``get_topic_kwargs``
+    to get the statistical data for the airTurbulence telemetry topic
+    and write the topic.
+
+    ``get_topic_kwargs`` also clears the accumulated data,
```

Comment:
I see that `get_topic_kwargs` throwns a RuntimeError when called too soon. Isn't it possible to find a more fail-safe implementation that won't involve throwing a RuntimeError? I'd hate to see a badly coded CSC go to FAULT state all the time.

Reply:
OK. I'll change the API to eliminate the `do_report` property and make `get_topic_kwargs` to an empty dict if there's not enough data accumulated. That simplifies the API, so it's probably an improvement.